base/sparse/sparsematrix.jl

```diff
@@ -1269,134 +1269,116 @@ end
+floor{To}(::Type{To}, A::SparseMatrixCSC) = _broadcast_unary_nz2z_z2z_T(floor, A, To)
+trunc{To}(::Type{To}, A::SparseMatrixCSC) = _broadcast_unary_nz2z_z2z_T(trunc, A, To)
+round{To}(::Type{To}, A::SparseMatrixCSC) = _broadcast_unary_nz2z_z2z_T(round, A, To)
+
+# Operations that map zeros to zeros and map nonzeros to nonzeros, yielding a sparse matrix
+"""
+Takes unary function `f` that maps zeros to zeros and nonzeros to nonzeros, and returns a
+new `SparseMatrixCSC{TiA,TvB}` `B` generated by applying `f` to each nonzero entry in `A`.
+"""
+function _broadcast_unary_nz2nz_z2z_T{TvA,TiA,TvB}(f::Function, A::SparseMatrixCSC{TvA,TiA}, ::Type{TvB})
+    Bcolptr = Vector{TiA}(A.n + 1)
+    Browval = Vector{TiA}(nnz(A))
+    Bnzval = Vector{TvB}(nnz(A))
+    copy!(Bcolptr, 1, A.colptr, 1, A.n + 1)
+    copy!(Browval, 1, A.rowval, 1, nnz(A))
+    @inbounds @simd for k in 1:nnz(A)
+        Bnzval[k] = f(A.nzval[k])
+    end
+    return SparseMatrixCSC(A.m, A.n, Bcolptr, Browval, Bnzval)
+end
+function _broadcast_unary_nz2nz_z2z{Tv}(f::Function, A::SparseMatrixCSC{Tv})
+    _broadcast_unary_nz2nz_z2z_T(f, A, Tv)
+end
+@_enumerate_childmethods(_broadcast_unary_nz2nz_z2z,
+    log1p, expm1, abs, abs2, conj)
+broadcast{TTv}(::typeof(abs2), A::SparseMatrixCSC{Complex{TTv}}) = _broadcast_unary_nz2nz_z2z_T(abs2, A, TTv)
+broadcast{TTv}(::typeof(abs), A::SparseMatrixCSC{Complex{TTv}}) = _broadcast_unary_nz2nz_z2z_T(abs, A, TTv)
+broadcast{TTv<:Integer}(::typeof(abs), A::SparseMatrixCSC{Complex{TTv}}) = _broadcast_unary_nz2nz_z2z_T(abs, A, Float64)
+broadcast{TTv<:BigInt}(::typeof(abs), A::SparseMatrixCSC{Complex{TTv}}) = _broadcast_unary_nz2nz_z2z_T(abs, A, BigFloat)
 function conj!(A::SparseMatrixCSC)
-    nzvalA = A.nzval
-    @simd for i=1:length(nzvalA)
-        @inbounds nzvalA[i] = conj(nzvalA[i])
+    @inbounds @simd for k in 1:nnz(A)
+        A.nzval[k] = conj(A.nzval[k])
     end
     return A
 end
 
-conj(A::SparseMatrixCSC) = conj!(copy(A))
-
-# Operations that map nonzeros to nonzeros, and zeros to nonzeros
-# Result is dense
-for op in (:cos, :cosh, :acos, :sec, :csc, :cot, :acot, :sech,
-           :csch, :coth, :asech, :acsch, :cospi, :sinc, :cosd,
-           :cotd, :cscd, :secd, :acosd, :acotd, :log, :log2, :log10,
-           :exp, :exp2, :exp10)
-    @eval begin
-
-        function ($op){Tv}(A::SparseMatrixCSC{Tv})
-            B = fill($(op)(zero(Tv)), size(A))
-            @inbounds for col = 1 : A.n
-                for j = A.colptr[col] : A.colptr[col+1]-1
-                    row = A.rowval[j]
-                    nz = A.nzval[j]
-                    B[row,col] = $(op)(nz)
-                end
-            end
-            return B
+# Operations that map both zeros and nonzeros to zeros, yielding a dense matrix
+"""
+Takes unary function `f` that maps both zeros and nonzeros to nonzeros, and returns a new
+`Matrix{TvB}` `B` effectively generated by applying `f` to every entry in `A`.
```

Comment:
Due to the `B = fill(f(zero(Tv)), size(A))`, this method calls `f` only once rather than once for each zero in `A`. Would make a difference if e.g. `f` has side effects.